torch/csrc/jit/mobile/promoted_prim_ops.cpp

```diff
@@ -0,0 +1,190 @@
+    out_ty = BoolType::get();
+  } else if (elem_ty_val == 3) {
+    out_ty = ComplexType::get();
+  } else {
+    TORCH_CHECK(
+        false,
+        "Unsupported element type for tolist; only int, float, complex and bool are supported");
+  }
+
+  // Check that type of the Tensor matches that of the annotation.
+  // Make an exception for the case in which the annotated type is
+  // float/complex and the Tensor data type is also float/complex;
+  // the elements will be casted to double/c10::complex<double>
+  // later.
+  TORCH_CHECK(
+      (out_ty == FloatType::get() && t.is_floating_point()) ||
+          (out_ty == ComplexType::get() && t.is_complex()) ||
+          tryScalarTypeFromJitType(out_ty) == t.scalar_type(),
+      "Output annotation element type and runtime tensor element type must match for tolist()");
+
+  // Check that the dimension of the Tensor matches that of the
+  // annotation.
+  TORCH_CHECK(
+      dim_val == t.dim(),
+      "Output annotation list dimension and runtime tensor dimension must match for tolist()");
+
+  // Wrap out_ty in a ListType dim times.
+  for (const auto i : c10::irange(dim_val)) {
+    (void)i; // Suppress unused variable warning
+    out_ty = ListType::create(out_ty);
+  }
+
+  int64_t dim = t.dim();
+  auto sizes = t.sizes();
+  auto strides = t.strides();
+  size_t element_size = t.element_size();
+  char* data = static_cast<char*>(t.data_ptr());
+  auto result = tensorToListRecursive(
+      data, 0, dim, out_ty, t.scalar_type(), sizes, strides, element_size);
+  push(stack, std::move(result));
+}
+
+void numToTensorScalar(Stack& stack) {
+  at::Scalar s;
+  pop(stack, s);
+  push(stack, at::scalar_to_tensor(s));
+}
+
+void isCuda(Stack& stack) {
+  at::Tensor a;
+  pop(stack, a);
+  push(stack, a.is_cuda());
+}
+
+void numToTensorBool(Stack& stack) {
+  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
+  bool b;
+  pop(stack, b);
+  push(stack, at::scalar_to_tensor(b));
+}
+
+static const mobile::prim_op_fn_register op_regs[] = {
+    mobile::prim_op_fn_register("prim::TupleIndex", tupleIndex),
+    mobile::prim_op_fn_register("aten::Bool.Tensor", boolTensor),
+    mobile::prim_op_fn_register("aten::format", aten_format),
+    mobile::prim_op_fn_register("prim::NumToTensor.Scalar", numToTensorScalar),
+    mobile::prim_op_fn_register("prim::RaiseException", raiseException),
+    mobile::prim_op_fn_register("prim::device", device),
+    mobile::prim_op_fn_register("prim::dtype", dtype),
+    mobile::prim_op_fn_register("aten::__not__", _not),
+    mobile::prim_op_fn_register("aten::__is__", is),
+    mobile::prim_op_fn_register("aten::__isnot__", isNot),
+    mobile::prim_op_fn_register("aten::dim", dim),
+    mobile::prim_op_fn_register("prim::Uninitialized", unInitialized),
+    mobile::prim_op_fn_register("aten::to.prim_dtype", toPrimDType),
+    mobile::prim_op_fn_register("prim::is_cuda", isCuda)
+    // TODO: (@pavithran) size is overloaded with int[] and Tensor
```

Comment:
It may be some legacy code. There are two operators: `aten::size` and `aten::Size`. The only difference is the cap `S`. You may need to look at which one need to be promoted.